DigitalLearningSolutions.Data/Services/TrackerActionService.cs

```diff
@@ -106,5 +117,57 @@ public TrackerEndpointResponse StoreDiagnosticJson(int? progressId, string? diag
 
             return TrackerEndpointResponse.Success;
         }
+
+        public TrackerEndpointResponse StoreAspProgressV2(
+            int? progressId,
+            int version,
+            string? progressText,
+            int tutorialId,
+            int? tutorialTime,
+            int? tutorialStatus,
+            int candidateId,
+            int? customisationId
+        )
+        {
+            if (progressId == null)
+            {
+                return TrackerEndpointResponse.StoreAspProgressV2Exception;
+            }
+
+            var progress = progressService.GetDetailedCourseProgress((int)progressId);
+            if (progress == null || progress.DelegateId != candidateId || progress.CustomisationId != customisationId)
+            {
+                return TrackerEndpointResponse.StoreAspProgressV2Exception;
+            }
+
+            if (tutorialStatus == null || tutorialTime == null)
+            {
+                return TrackerEndpointResponse.NullTutorialStatusOrTime;
+            }
+
+            try
+            {
+                progressService.StoreAspProgressV2(
+                    progress.ProgressId,
+                    version,
+                    progressText,
+                    tutorialId,
+                    (int)tutorialTime,
+                    (int)tutorialStatus

```

Comment:
Is there any particular reason you'be done this this way rather than the usual tutorialTime!.Value that we've been using elsewhere?

Reply:
Ah cool I can do it that way